### PR TITLE
Unit 13 : Outlook - Hyena : fixing a typo in the hyena chapter + improvements

### PR DESCRIPTION
**Pull request by lulmer**
This last minute PR aims to fix a typo in the hyena chapter
Along the way :  

- improving a fuzy explanation on the conv - toeplitz relationship and removing a confusing schema
- adding tips with notions to make it easier to understand
- adding an extra paper that is an improvement of hyena given as an outlook for the reader  
- and some real application examples
- formating to get to 100 lines ;)

### CODE REVIEW

**Review by MKhalusova — COMMENTED** [review]
This is a very important part of the course, but I feel that it might be a bit difficult to digest, as the writing style is a bit dry and very academic. Could you please try to make it a bit more approachable for an average learner?

File: chapters/en/Unit 13 - Outlook/hyena.mdx [hunk 1]
```diff
@@ -27,21 +32,29 @@ In the realm of computational efficiency, the Hyena operator attains an evaluati
 Let's delve into the second-order recursion of the Hyena operator, which simplifies its representation for illustrative purposes.
 ![hyena_mechanism.png](https://huggingface.co/datasets/hf-vision/course-assets/resolve/main/outlook_hyena_images/hyena_mechanism.png)
 
-In this context, $x_1$ and $x_2$ are equivalent to $q$ and $v$ attention vectors from the Attention mechanism. However, unlike the attention mechanism, which typically uses a dense layer for projecting the input sequence into representations, Hyena incorporates both a dense layer and standard convolutions. Here, $k$ and $v$ are aggregated with an elementwise multiplication operation. 
+In this context, $x_1$ and $x_2$ are equivalent to $q$ and $v$ attention vectors from the Attention mechanism.
```
[the remark this answers is not in the document]
Comment (lulmer) [reply chapters/en/Unit 13 - Outlook/hyena.mdx:35]: There is already an image that shows that Attention and Hyena layers have the same positioning within a typical "transformer block". But I see what you mean, will add something detailed 👍

File: chapters/en/Unit 13 - Outlook/hyena.mdx [hunk 2]
```diff
@@ -1,13 +1,18 @@
+# Hyena 
+
 ## Overview
 
 ### What is Hyena 
 
 Hyena is a new type of operator that serves as a substitute for the attention mechanism. Developed by Hazy Research, it features a subquadratic computational efficiency, constructed by interleaving implicitly parametrized long convolutions and data-controlled gating.
```
[the remark this answers is not in the document]
Comment (lulmer) [reply chapters/en/Unit 13 - Outlook/hyena.mdx:7]: Thank you for reviewing ! It is super helpful 👍 
My understanding is that Hyena is not an acronym that as a particular explanation... Can you be more specific on your expectations ?

File: chapters/en/Unit 13 - Outlook/hyena.mdx [hunk 3]
```diff
@@ -51,33 +64,40 @@ In essence, Hyena can be performed in two steps:
 
 ## Why it matters
 
-While Transformer is a well established and very capable architecture, the quadratic computational cost is an expensive price to pay, especially in inference. The H3 mechanism proposition went close to the perplexity of multi-headed attention mechanisms, but there was still a narrow gap in terms of perplexity. 
+While Transformer is a well established and very capable architecture, the quadratic computational cost is an expensive price to pay, especially in inference.
```
[the remark this answers is not in the document]
Comment (lulmer) [reply chapters/en/Unit 13 - Outlook/hyena.mdx:67]: You are right, I will do it

File: chapters/en/Unit 13 - Outlook/hyena.mdx [hunk 4]
```diff
@@ -27,21 +32,29 @@ In the realm of computational efficiency, the Hyena operator attains an evaluati
...
-This transformation leads to greater parameter efficiency. Instead of directly learning fixed kernel weight values, a parametrized function is employed. This function intelligently deduces the values of the kernel weights and their dimensions during the network's forward pass, optimizing resource use.
+This transformation leads to greater parameter efficiency. 
+Instead of directly learning fixed kernel weight values, a parametrized function is employed. 
+This function intelligently deduces the values of the kernel weights and their dimensions during the network's forward pass, optimizing resource use.
 
-In practice, convolutions are accelerated to a subquadratic time complexity by the Cooley-Tukey fast Fourier transform (FFT) algorithm. Some work has been conducted to speed up this computation like FastFFTConv based on Monarch decomposition. 
+In practice, convolutions are accelerated to a subquadratic time complexity by the Cooley-Tukey fast Fourier transform (FFT) algorithm. 
+Some work has been conducted to speed up this computation like FastFFTConv based on Monarch decomposition. 
 ### Wrapping up everything
```
Comment (merveenoyan) [inline chapters/en/Unit 13 - Outlook/hyena.mdx:58]:
```suggestion
### Wrapping Up Everything
```

File: chapters/en/Unit 13 - Outlook/hyena.mdx [hunk 5]
```diff
@@ -51,33 +64,40 @@ In essence, Hyena can be performed in two steps:
 
 ## Why it matters
```
Comment (merveenoyan) [inline chapters/en/Unit 13 - Outlook/hyena.mdx:65]:
```suggestion
## Why Hyena Matters
```

File: chapters/en/Unit 13 - Outlook/hyena.mdx [hunk 6]
```diff
@@ -51,33 +64,40 @@ In essence, Hyena can be performed in two steps:
...
-Hyena Hierarchy facilitates the development of larger, more efficient convolution models for long sequences. The potential for Hyena type models for computer vision would be a more efficient GPU memory consumption of patches, that would allow : 
+Hyena Hierarchy facilitates the development of larger, more efficient convolution models for long sequences. 
+The potential for Hyena type models for computer vision would be a more efficient GPU memory consumption of patches, that would allow : 
 * The processing of larger, higher-resolution images
-* The use ofwith smaller patches, allowing a fine-graine feature representation
+* The use of smaller patches, allowing a fine-graine feature representation 
+
+These qualities would be particularly beneficial in areas such as Medical Imaging and Remote Sensing.
+
 ## Toward Transformers Alternatives
```
Comment (merveenoyan) [inline chapters/en/Unit 13 - Outlook/hyena.mdx:91]:
```suggestion
## Towards Transformers Alternatives 
```

File: chapters/en/Unit 13 - Outlook/hyena.mdx [hunk 7]
```diff
@@ -51,33 +64,40 @@ In essence, Hyena can be performed in two steps:
...
+These qualities would be particularly beneficial in areas such as Medical Imaging and Remote Sensing.
+
 ## Toward Transformers Alternatives 
 Building new layers from simple design principles is an emerging research field that is progressing very quickly. 
 
-The H3 mechanism serves as the foundation for many State Space Model based (SSM) architectures, typically featuring a structure that alternates between a block inspired by linear attention and a multi-layer perceptron (MLP) block. Hyena, as an enhancement of this approach, has paved the way for even more efficient architectures such as Mamba.
+The H3 mechanism serves as the foundation for many State Space Model based (SSM) architectures, typically featuring a structure that alternates between a block inspired by linear attention and a multi-layer perceptron (MLP) block. 
+Hyena, as an enhancement of this approach, has paved the way for even more efficient architectures such as Mamba and its derivatives for vision (Vision Mamba, VMamba etc...).
 
 ## Further reading
```
Comment (merveenoyan) [inline chapters/en/Unit 13 - Outlook/hyena.mdx:97]:
```suggestion
## Further Reading
```

**Comment by SolshineCode** [comment]
<p>Good catch on the typo.</p>

**Comment by lulmer** [comment]
Added a new commit that takes into account @MKhalusova suggestions to make it more accessible for learners. Hope it is better now !

**Review by MKhalusova — APPROVED** [review]
Thank you for iterating on the style! I think it should be much more accessible to learners now.

**Pull request merged**